### PR TITLE
feat(iOS): add support for missing text content types

### PR DESCRIPTION
**Pull request by krozniata**
## Summary:

This PR adds few missing text content types on iOS (available from iOS 15)

- dateTime
- flightNumber
- shipmentTrackingNumber

## Changelog:

[IOS] [ADDED] - Add support for missing text content types

## Test Plan:

Make sure that `RNTester` builds and runs successfully

### CODE REVIEW

File: packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.mm [hunk 1]
```diff
@@ -314,6 +314,16 @@ - (void)setTextContentType:(NSString *)type
...
+      
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_15_0
+    if (@available(iOS 15.0, *)) {
+      [mutableContentTypeMap addEntriesFromDictionary:@{
+        @"dateTime" : UITextContentTypeDateTime,
+        @"flightNumber" : UITextContentTypeFlightNumber,
+        @"shipmentTrackingNumber" : UITextContentTypeShipmentTrackingNumber,
+      }];
+    }
+#endif
```
Comment (javache) [inline packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.mm:326]: Can we order these from old to new (so first IOS15+, then iOS17+). 

Also, we're always targeting the latest iOS SDK, so we don't need these ifdefs anymore I beleive.

**Review by javache — CHANGES_REQUESTED** [review]

File: packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.mm (hunk 1 above)
Comment (krozniata) [reply packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.mm]: @javache 

I changed order and removed ifdefs only for iOS15 as removing those for iOS17 was causing pipeline to fail

**Comment by krozniata** [comment]
Hey @javache, I've updated this PR according to your suggestions, is there anything that needs to be done to get it merged?

**Comment by krozniata** [comment]
@cipolleschi 👀

**Review by cipolleschi — APPROVED** [review]
Thanks for adding these. How do they work on Android? If I set `dateTime` as `textContentType`, how does Android interpret it?

**Comment by cipolleschi** [comment]
Can you also rebase on top of master? I can't see the GitHub Actions running

**Comment by krozniata** [comment]
@cipolleschi 

rebased 👌

`textContentType` is iOS only, so it won't have any effect on Android

**Pull request closed without merging**